File: gluecodium/src/test/resources/smoke/equatable/output/cpp/src/smoke/Equatable.cpp
// -------------------------------------------------------------------------------------------------
//
//
// -------------------------------------------------------------------------------------------------
#include "smoke/Equatable.h"
#include <utility>
namespace smoke {
Equatable::EquatableStruct::EquatableStruct( )
    : bool_field{ }, int_field{ }, long_field{ }, float_field{ }, double_field{ }, string_field{ }, struct_field{ }, enum_field{ }, array_field{ }, map_field{ }
{
}
Equatable::EquatableStruct::EquatableStruct( bool bool_field, int32_t int_field, int64_t long_field, float float_field, double double_field, ::std::string string_field, ::smoke::Equatable::NestedEquatableStruct struct_field, ::smoke::Equatable::SomeEnum enum_field, ::std::vector< ::std::string > array_field, ::smoke::Equatable::ErrorCodeToMessageMap map_field )
    : bool_field( std::move( bool_field ) ), int_field( std::move( int_field ) ), long_field( std::move( long_field ) ), float_field( std::move( float_field ) ), double_field( std::move( double_field ) ), string_field( std::move( string_field ) ), struct_field( std::move( struct_field ) ), enum_field( std::move( enum_field ) ), array_field( std::move( array_field ) ), map_field( std::move( map_field ) )
{
}
bool Equatable::EquatableStruct::operator==( const EquatableStruct& rhs ) const
{
    return bool_field == rhs.bool_field &&
        int_field == rhs.int_field &&
        long_field == rhs.long_field &&
        float_field == rhs.float_field &&
        double_field == rhs.double_field &&
        string_field == rhs.string_field &&
        struct_field == rhs.struct_field &&
        enum_field == rhs.enum_field &&
        array_field == rhs.array_field &&
        map_field == rhs.map_field;
}
bool Equatable::EquatableStruct::operator!=( const EquatableStruct& rhs ) const
{
    return !( *this == rhs );
}
Equatable::EquatableNullableStruct::EquatableNullableStruct( )
    : bool_field{ }, int_field{ }, uint_field{ }, float_field{ }, string_field{ }, struct_field{ }, enum_field{ }, array_field{ }, map_field{ }
{
}
Equatable::EquatableNullableStruct::EquatableNullableStruct( ::gluecodium::optional< bool > bool_field, ::gluecodium::optional< int32_t > int_field, ::gluecodium::optional< uint16_t > uint_field, ::gluecodium::optional< float > float_field, ::gluecodium::optional< ::std::string > string_field, ::gluecodium::optional< ::smoke::Equatable::NestedEquatableStruct > struct_field, ::gluecodium::optional< ::smoke::Equatable::SomeEnum > enum_field, ::gluecodium::optional< ::std::vector< ::std::string > > array_field, ::gluecodium::optional< ::smoke::Equatable::ErrorCodeToMessageMap > map_field )
    : bool_field( std::move( bool_field ) ), int_field( std::move( int_field ) ), uint_field( std::move( uint_field ) ), float_field( std::move( float_field ) ), string_field( std::move( string_field ) ), struct_field( std::move( struct_field ) ), enum_field( std::move( enum_field ) ), array_field( std::move( array_field ) ), map_field( std::move( map_field ) )
{
}
bool Equatable::EquatableNullableStruct::operator==( const EquatableNullableStruct& rhs ) const
{
    return ( ( bool_field && rhs.bool_field )
            ? ( *bool_field == *rhs.bool_field )
            : ( static_cast< bool >( bool_field ) == static_cast< bool >( rhs.bool_field ) ) ) &&
        ( ( int_field && rhs.int_field )
            ? ( *int_field == *rhs.int_field )
            : ( static_cast< bool >( int_field ) == static_cast< bool >( rhs.int_field ) ) ) &&
        ( ( uint_field && rhs.uint_field )
            ? ( *uint_field == *rhs.uint_field )
            : ( static_cast< bool >( uint_field ) == static_cast< bool >( rhs.uint_field ) ) ) &&
        ( ( float_field && rhs.float_field )
            ? ( *float_field == *rhs.float_field )
            : ( static_cast< bool >( float_field ) == static_cast< bool >( rhs.float_field ) ) ) &&
        ( ( string_field && rhs.string_field )
            ? ( *string_field == *rhs.string_field )
            : ( static_cast< bool >( string_field ) == static_cast< bool >( rhs.string_field ) ) ) &&
        ( ( struct_field && rhs.struct_field )
            ? ( *struct_field == *rhs.struct_field )
            : ( static_cast< bool >( struct_field ) == static_cast< bool >( rhs.struct_field ) ) ) &&
        ( ( enum_field && rhs.enum_field )
            ? ( *enum_field == *rhs.enum_field )
            : ( static_cast< bool >( enum_field ) == static_cast< bool >( rhs.enum_field ) ) ) &&
        ( ( array_field && rhs.array_field )
            ? ( *array_field == *rhs.array_field )
            : ( static_cast< bool >( array_field ) == static_cast< bool >( rhs.array_field ) ) ) &&
        ( ( map_field && rhs.map_field )
            ? ( *map_field == *rhs.map_field )
            : ( static_cast< bool >( map_field ) == static_cast< bool >( rhs.map_field ) ) );
}
bool Equatable::EquatableNullableStruct::operator!=( const EquatableNullableStruct& rhs ) const
{
    return !( *this == rhs );
}
Equatable::NestedEquatableStruct::NestedEquatableStruct( )
    : foo_field{ }
{
}
Equatable::NestedEquatableStruct::NestedEquatableStruct( ::std::string foo_field )
    : foo_field( std::move( foo_field ) )
{
}
bool Equatable::NestedEquatableStruct::operator==( const NestedEquatableStruct& rhs ) const
{
    return foo_field == rhs.foo_field;
}
bool Equatable::NestedEquatableStruct::operator!=( const NestedEquatableStruct& rhs ) const
{
    return !( *this == rhs );
}
}
namespace gluecodium {
std::size_t
hash< ::smoke::Equatable::EquatableStruct >::operator( )( const ::smoke::Equatable::EquatableStruct& t ) const
{
    size_t hash_value = 43;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::bool_field ) >( )( t.bool_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::int_field ) >( )( t.int_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::long_field ) >( )( t.long_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::float_field ) >( )( t.float_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::double_field ) >( )( t.double_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::string_field ) >( )( t.string_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::struct_field ) >( )( t.struct_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::enum_field ) >( )( t.enum_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::array_field ) >( )( t.array_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableStruct::map_field ) >( )( t.map_field ) ) << 1;
    return hash_value;
}
std::size_t
hash< ::smoke::Equatable::EquatableNullableStruct >::operator( )( const ::smoke::Equatable::EquatableNullableStruct& t ) const
{
    size_t hash_value = 43;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::bool_field ) >( )( t.bool_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::int_field ) >( )( t.int_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::uint_field ) >( )( t.uint_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::float_field ) >( )( t.float_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::string_field ) >( )( t.string_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::struct_field ) >( )( t.struct_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::enum_field ) >( )( t.enum_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::array_field ) >( )( t.array_field ) ) << 1;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::EquatableNullableStruct::map_field ) >( )( t.map_field ) ) << 1;
    return hash_value;
}
std::size_t
hash< ::smoke::Equatable::NestedEquatableStruct >::operator( )( const ::smoke::Equatable::NestedEquatableStruct& t ) const
{
    size_t hash_value = 43;
    hash_value = ( hash_value ^ ::gluecodium::hash< decltype( ::smoke::Equatable::NestedEquatableStruct::foo_field ) >( )( t.foo_field ) ) << 1;
    return hash_value;
}
}

File: gluecodium/src/test/resources/smoke/equatable/output/swift/smoke/Equatable.swift
//
//
import Foundation
public struct Equatable {
    public enum SomeEnum : UInt32, CaseIterable, Codable {
        case foo
        case bar
    }
    public typealias ErrorCodeToMessageMap = [Int32: String]
    public struct EquatableStruct: Hashable {
        public var boolField: Bool
        public var intField: Int32
        public var longField: Int64
        public var floatField: Float
        public var doubleField: Double
        public var stringField: String
        public var structField: Equatable.NestedEquatableStruct
        public var enumField: Equatable.SomeEnum
        public var arrayField: [String]
        public var mapField: Equatable.ErrorCodeToMessageMap
        public init(boolField: Bool, intField: Int32, longField: Int64, floatField: Float, doubleField: Double, stringField: String, structField: Equatable.NestedEquatableStruct, enumField: Equatable.SomeEnum, arrayField: [String], mapField: Equatable.ErrorCodeToMessageMap) {
            self.boolField = boolField
            self.intField = intField
            self.longField = longField
            self.floatField = floatField
            self.doubleField = doubleField
            self.stringField = stringField
            self.structField = structField
            self.enumField = enumField
            self.arrayField = arrayField
            self.mapField = mapField
        }
        internal init(cHandle: _baseRef) {
            boolField = moveFromCType(smoke_Equatable_EquatableStruct_boolField_get(cHandle))
            intField = moveFromCType(smoke_Equatable_EquatableStruct_intField_get(cHandle))
            longField = moveFromCType(smoke_Equatable_EquatableStruct_longField_get(cHandle))
            floatField = moveFromCType(smoke_Equatable_EquatableStruct_floatField_get(cHandle))
            doubleField = moveFromCType(smoke_Equatable_EquatableStruct_doubleField_get(cHandle))
            stringField = moveFromCType(smoke_Equatable_EquatableStruct_stringField_get(cHandle))
            structField = moveFromCType(smoke_Equatable_EquatableStruct_structField_get(cHandle))
            enumField = moveFromCType(smoke_Equatable_EquatableStruct_enumField_get(cHandle))
            arrayField = foobar_moveFromCType(smoke_Equatable_EquatableStruct_arrayField_get(cHandle))
            mapField = foobar_moveFromCType(smoke_Equatable_EquatableStruct_mapField_get(cHandle))
        }
    }
    public struct EquatableNullableStruct: Hashable {
        public var boolField: Bool?
        public var intField: Int32?
        public var uintField: UInt16?
        public var floatField: Float?
        public var stringField: String?
        public var structField: Equatable.NestedEquatableStruct?
        public var enumField: Equatable.SomeEnum?
        public var arrayField: [String]?
        public var mapField: Equatable.ErrorCodeToMessageMap?
        public init(boolField: Bool? = nil, intField: Int32? = nil, uintField: UInt16? = nil, floatField: Float? = nil, stringField: String? = nil, structField: Equatable.NestedEquatableStruct? = nil, enumField: Equatable.SomeEnum? = nil, arrayField: [String]? = nil, mapField: Equatable.ErrorCodeToMessageMap? = nil) {
            self.boolField = boolField
            self.intField = intField
            self.uintField = uintField
            self.floatField = floatField
            self.stringField = stringField
            self.structField = structField
            self.enumField = enumField
            self.arrayField = arrayField
            self.mapField = mapField
        }
        internal init(cHandle: _baseRef) {
            boolField = moveFromCType(smoke_Equatable_EquatableNullableStruct_boolField_get(cHandle))
            intField = moveFromCType(smoke_Equatable_EquatableNullableStruct_intField_get(cHandle))
            uintField = moveFromCType(smoke_Equatable_EquatableNullableStruct_uintField_get(cHandle))
            floatField = moveFromCType(smoke_Equatable_EquatableNullableStruct_floatField_get(cHandle))
            stringField = moveFromCType(smoke_Equatable_EquatableNullableStruct_stringField_get(cHandle))
            structField = moveFromCType(smoke_Equatable_EquatableNullableStruct_structField_get(cHandle))
            enumField = moveFromCType(smoke_Equatable_EquatableNullableStruct_enumField_get(cHandle))
            arrayField = foobar_moveFromCType(smoke_Equatable_EquatableNullableStruct_arrayField_get(cHandle))
            mapField = foobar_moveFromCType(smoke_Equatable_EquatableNullableStruct_mapField_get(cHandle))
        }
    }
    public struct NestedEquatableStruct: Hashable {
        public var fooField: String
        public init(fooField: String) {
            self.fooField = fooField
        }
        internal init(cHandle: _baseRef) {
            fooField = moveFromCType(smoke_Equatable_NestedEquatableStruct_fooField_get(cHandle))
        }
    }
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.EquatableStruct {
    return Equatable.EquatableStruct(cHandle: handle)
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.EquatableStruct {
    defer {
        smoke_Equatable_EquatableStruct_release_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.EquatableStruct) -> RefHolder {
    let c_boolField = moveToCType(swiftType.boolField)
    let c_intField = moveToCType(swiftType.intField)
    let c_longField = moveToCType(swiftType.longField)
    let c_floatField = moveToCType(swiftType.floatField)
    let c_doubleField = moveToCType(swiftType.doubleField)
    let c_stringField = moveToCType(swiftType.stringField)
    let c_structField = moveToCType(swiftType.structField)
    let c_enumField = moveToCType(swiftType.enumField)
    let c_arrayField = foobar_moveToCType(swiftType.arrayField)
    let c_mapField = foobar_moveToCType(swiftType.mapField)
    return RefHolder(smoke_Equatable_EquatableStruct_create_handle(c_boolField.ref, c_intField.ref, c_longField.ref, c_floatField.ref, c_doubleField.ref, c_stringField.ref, c_structField.ref, c_enumField.ref, c_arrayField.ref, c_mapField.ref))
}
internal func moveToCType(_ swiftType: Equatable.EquatableStruct) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_EquatableStruct_release_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.EquatableStruct? {
    guard handle != 0 else {
        return nil
    }
    let unwrappedHandle = smoke_Equatable_EquatableStruct_unwrap_optional_handle(handle)
    return Equatable.EquatableStruct(cHandle: unwrappedHandle) as Equatable.EquatableStruct
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.EquatableStruct? {
    defer {
        smoke_Equatable_EquatableStruct_release_optional_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.EquatableStruct?) -> RefHolder {
    guard let swiftType = swiftType else {
        return RefHolder(0)
    }
    let c_boolField = moveToCType(swiftType.boolField)
    let c_intField = moveToCType(swiftType.intField)
    let c_longField = moveToCType(swiftType.longField)
    let c_floatField = moveToCType(swiftType.floatField)
    let c_doubleField = moveToCType(swiftType.doubleField)
    let c_stringField = moveToCType(swiftType.stringField)
    let c_structField = moveToCType(swiftType.structField)
    let c_enumField = moveToCType(swiftType.enumField)
    let c_arrayField = foobar_moveToCType(swiftType.arrayField)
    let c_mapField = foobar_moveToCType(swiftType.mapField)
    return RefHolder(smoke_Equatable_EquatableStruct_create_optional_handle(c_boolField.ref, c_intField.ref, c_longField.ref, c_floatField.ref, c_doubleField.ref, c_stringField.ref, c_structField.ref, c_enumField.ref, c_arrayField.ref, c_mapField.ref))
}
internal func moveToCType(_ swiftType: Equatable.EquatableStruct?) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_EquatableStruct_release_optional_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.EquatableNullableStruct {
    return Equatable.EquatableNullableStruct(cHandle: handle)
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.EquatableNullableStruct {
    defer {
        smoke_Equatable_EquatableNullableStruct_release_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.EquatableNullableStruct) -> RefHolder {
    let c_boolField = moveToCType(swiftType.boolField)
    let c_intField = moveToCType(swiftType.intField)
    let c_uintField = moveToCType(swiftType.uintField)
    let c_floatField = moveToCType(swiftType.floatField)
    let c_stringField = moveToCType(swiftType.stringField)
    let c_structField = moveToCType(swiftType.structField)
    let c_enumField = moveToCType(swiftType.enumField)
    let c_arrayField = foobar_moveToCType(swiftType.arrayField)
    let c_mapField = foobar_moveToCType(swiftType.mapField)
    return RefHolder(smoke_Equatable_EquatableNullableStruct_create_handle(c_boolField.ref, c_intField.ref, c_uintField.ref, c_floatField.ref, c_stringField.ref, c_structField.ref, c_enumField.ref, c_arrayField.ref, c_mapField.ref))
}
internal func moveToCType(_ swiftType: Equatable.EquatableNullableStruct) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_EquatableNullableStruct_release_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.EquatableNullableStruct? {
    guard handle != 0 else {
        return nil
    }
    let unwrappedHandle = smoke_Equatable_EquatableNullableStruct_unwrap_optional_handle(handle)
    return Equatable.EquatableNullableStruct(cHandle: unwrappedHandle) as Equatable.EquatableNullableStruct
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.EquatableNullableStruct? {
    defer {
        smoke_Equatable_EquatableNullableStruct_release_optional_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.EquatableNullableStruct?) -> RefHolder {
    guard let swiftType = swiftType else {
        return RefHolder(0)
    }
    let c_boolField = moveToCType(swiftType.boolField)
    let c_intField = moveToCType(swiftType.intField)
    let c_uintField = moveToCType(swiftType.uintField)
    let c_floatField = moveToCType(swiftType.floatField)
    let c_stringField = moveToCType(swiftType.stringField)
    let c_structField = moveToCType(swiftType.structField)
    let c_enumField = moveToCType(swiftType.enumField)
    let c_arrayField = foobar_moveToCType(swiftType.arrayField)
    let c_mapField = foobar_moveToCType(swiftType.mapField)
    return RefHolder(smoke_Equatable_EquatableNullableStruct_create_optional_handle(c_boolField.ref, c_intField.ref, c_uintField.ref, c_floatField.ref, c_stringField.ref, c_structField.ref, c_enumField.ref, c_arrayField.ref, c_mapField.ref))
}
internal func moveToCType(_ swiftType: Equatable.EquatableNullableStruct?) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_EquatableNullableStruct_release_optional_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.NestedEquatableStruct {
    return Equatable.NestedEquatableStruct(cHandle: handle)
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.NestedEquatableStruct {
    defer {
        smoke_Equatable_NestedEquatableStruct_release_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.NestedEquatableStruct) -> RefHolder {
    let c_fooField = moveToCType(swiftType.fooField)
    return RefHolder(smoke_Equatable_NestedEquatableStruct_create_handle(c_fooField.ref))
}
internal func moveToCType(_ swiftType: Equatable.NestedEquatableStruct) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_NestedEquatableStruct_release_handle)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.NestedEquatableStruct? {
    guard handle != 0 else {
        return nil
    }
    let unwrappedHandle = smoke_Equatable_NestedEquatableStruct_unwrap_optional_handle(handle)
    return Equatable.NestedEquatableStruct(cHandle: unwrappedHandle) as Equatable.NestedEquatableStruct
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.NestedEquatableStruct? {
    defer {
        smoke_Equatable_NestedEquatableStruct_release_optional_handle(handle)
    }
    return copyFromCType(handle)
}
internal func copyToCType(_ swiftType: Equatable.NestedEquatableStruct?) -> RefHolder {
    guard let swiftType = swiftType else {
        return RefHolder(0)
    }
    let c_fooField = moveToCType(swiftType.fooField)
    return RefHolder(smoke_Equatable_NestedEquatableStruct_create_optional_handle(c_fooField.ref))
}
internal func moveToCType(_ swiftType: Equatable.NestedEquatableStruct?) -> RefHolder {
    return RefHolder(ref: copyToCType(swiftType).ref, release: smoke_Equatable_NestedEquatableStruct_release_optional_handle)
}
internal func copyToCType(_ swiftEnum: Equatable.SomeEnum) -> PrimitiveHolder<UInt32> {
    return PrimitiveHolder(swiftEnum.rawValue)
}
internal func moveToCType(_ swiftEnum: Equatable.SomeEnum) -> PrimitiveHolder<UInt32> {
    return copyToCType(swiftEnum)
}
internal func copyToCType(_ swiftEnum: Equatable.SomeEnum?) -> RefHolder {
    return copyToCType(swiftEnum?.rawValue)
}
internal func moveToCType(_ swiftEnum: Equatable.SomeEnum?) -> RefHolder {
    return moveToCType(swiftEnum?.rawValue)
}
internal func copyFromCType(_ cValue: UInt32) -> Equatable.SomeEnum {
    return Equatable.SomeEnum(rawValue: cValue)!
}
internal func moveFromCType(_ cValue: UInt32) -> Equatable.SomeEnum {
    return copyFromCType(cValue)
}
internal func copyFromCType(_ handle: _baseRef) -> Equatable.SomeEnum? {
    guard handle != 0 else {
        return nil
    }
    return Equatable.SomeEnum(rawValue: uint32_t_value_get(handle))!
}
internal func moveFromCType(_ handle: _baseRef) -> Equatable.SomeEnum? {
    defer {
        uint32_t_release_handle(handle)
    }
    return copyFromCType(handle)
}
